controllers/serviceexport/reconciler.go

```diff
@@ -62,18 +62,9 @@ var finalizerName = "networking.kubeslice.io/serviceexport-finalizer"
 func (r Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithValues("serviceexport", req.NamespacedName)
 
-	serviceexport := &kubeslicev1beta1.ServiceExport{}
-	err := r.Get(ctx, req.NamespacedName, serviceexport)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Return and don't requeue
-			log.Info("serviceexport resource not found. Ignoring since object must be deleted")
-			return ctrl.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
-		log.Error(err, "Failed to get serviceexport")
-		return ctrl.Result{}, err
+	serviceexport := r.GetServiceExport(ctx, req, &log)
+	if serviceexport == nil {
```

Comment:
Shouldn't this return an actual error, if serviceexport not found, like it was returning previously

Reply:
You are right, I was trying to make it too terse. Fixed it